### PR TITLE
Replace RequestResponseFactory by Sling Request builders

### LINKED ISSUE #3404: Replace deprecated com.day.cq.contentsync.handler.util.RequestResponseFactory by SlingHttpServletRequestBuilder
The package `com.day.cq.contentsync.handler.util` is deprecated in AEMaaCS (https://experienceleague.adobe.com/en/docs/experience-manager-cloud-service/content/release-notes/deprecated-removed-features). One should use https://sling.apache.org/apidocs/sling12/org/apache/sling/api/request/builder/SlingHttpServletRequestBuilder.html instead. This requires Sling API 2.24.0+ (satisfied by 6.5.21 or newer)

### PR DESCRIPTION
**Pull request by kwin**
This closes #3404

### CODE REVIEW

**Comment by kwin** [comment]
This fails because AEM 6.5.21 exports `com.day.cq.wcm.api` in version 1.31.0 (from bundle `com.day.cq.wcm.cq-wcm-api 
 5.14.18`)  while AEM SDK API (in newest version 2024.7.17258.20240726T172406Z-240700) only exports that package only in version 1.30.0. (despite the fact that this comes with bundle `com.day.cq.wcm.cq-wcm-api 5.15.28`)

@cziegeler Any idea when AEMaaCS is updated to a WCM API bundle which satisfies the import-package version ranges being generated with `uber-jar 6.5.21`?

**Comment by cziegeler** [comment]
Thanks for reporting, we will investigate.
As a workaround I guess we could explicitly specify a version range for that package that satisfies both?

**Comment by davidjgonzalez** [comment]
Will this up the required AEM 6.5 version that contains this release? (ie. back breaking change)

**Comment by kwin** [comment]
> Will this up the required AEM 6.5 version that contains this release? (ie. back breaking change)

Yes, requires AEM 6.5.21 (no update for AEM SDK required though)

**Review by davidjgonzalez — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,9 +28,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 - #3357 - Added debugging and null checking to ReferencesModel to prevent NPE
 - #3398 - CreateRedirectConfigurationServlet throws PersistenceException when ancestor node types are different than expected
 - #3275 - CCVAR: Fixed Same Attribute not updating correctly.
+- #3402 - EnsureOakIndexManagerImpl does not pick up changes in EnsureOakIndex configurations.
 
 ### Changed
 
+- #3403 - Replace deprecated com.day.cq.contentsync.handler.util.RequestResponseFactory by SlingHttpServletRequestBuilder
```
Comment (kwin) [inline CHANGELOG.md:35]: This is incorrect as 6.6.4 has been released already.

**Pull request merged**